src/Common/src/System/Security/Cryptography/Asn1Reader/AsnWriter.cs

```diff
@@ -57,7 +57,8 @@ public void Dispose()
             {
                 Array.Clear(_buffer, 0, _offset);
 #if !CHECK_ACCURATE_ENSURE
```

Comment:
In what situation is this constant set?  Are we leaking an array in that case?  Could we just always return the array via a call to `CryptoPool.Return(_buffer, _offset)` and remove the Clear call?

Reply:
It's just manually set, it's a testing tool to make sure that writing after calling EnsureWriteCapacity is not accidentally exceeding bounds, by making it not give more than was requested.

https://github.com/dotnet/corefx/blob/d8797a0f1616c1db1e5294964f2826e051f2615e/src/Common/src/System/Security/Cryptography/Asn1Reader/AsnWriter.cs#L214-L224